misc/factorial_recursive.c

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file factorial_recursive.c
+ * @brief factorial with recursive function call
+ * @details
+ * This version of the factorial computation function obtains the result by
+ * recursively calling the function
+ * @author [Nicola Masarone](https://github.com/nicola-masarone)
+ *
+ */
+
+#include <assert.h>
+#include <stdio.h>
```

Comment:
Add a brief description of what the library `stdio.h` adds (see the example below).
```suggestion
#include <assert.h>   /// for assert
#include <stdio.h>   /// for
```